### PR TITLE
Solve Task 6 CORE: GET /api/articles/:article_id/comments

### PR DESCRIPTION
**Pull request by isabelpopa**

### CODE REVIEW

File: controllers/comments.controllers.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
+const { request } = require("express");
```
Comment (Keb84) [inline controllers/comments.controllers.js]: You don't need express required into your controller. You are already requiring this into the app which is where these functions are being exported too 🙂

File: models/comments.models.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    .then(({ rows }) => {
+      const comment = rows[0];
+      if (!comment) {
+        return Promise.reject({
+          status: 404,
+          msg: "Article_id Not Found",
+        });
+      } else {
+        return rows;
+      }
```
Comment (Keb84) [inline models/comments.models.js]: What would happen here if the review id existed but there were no comments yet? 🤔 you would still get an empty array but the an error message of article_id not found would be the response?

File: __tests__/api/api.test.js [hunk 3]
```diff
@@ -166,6 +166,61 @@ describe("GET /api/articles", () => {
...
+      });
+  });
+  test("Comments should be sorted by most recent comments first.", () => {
+    return request(app)
+      .get("/api/articles/1/comments")
+      .then(({ body }) => {
+        const { comments } = body;
+        expect(comments).toBeSortedBy('created_at', {descending: true});
+      });
+  });
```
Comment (Keb84) [inline __tests__/api/api.test.js:212]: Another 200 status test to check if the id exits but the array is empty is needed here. See my comments in the models that relates to this 👀

File: __tests__/api/api.test.js [hunk 4]
```diff
@@ -166,6 +166,61 @@ describe("GET /api/articles", () => {
...
+            comment_id: 5,
+            body: "I hate streaming noses",
+            votes: 0,
+            author: "icellusedkars",
+            article_id: 1,
+            created_at: "2020-11-03T21:00:00.000Z",
+          });
+        });
+      });
+  });
```
Comment (B7Bprog) [inline __tests__/api/api.test.js:195]: Please add another test 200 where there are no comments for an existing article. This should respond with an empty array.

**Review by Keb84 — CHANGES_REQUESTED** [review]
Nice work so far 🤸 please see my comments and resubmit PR when done 👍

File: controllers/comments.controllers.js [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+const { selectCommentsByArticleId } = require("../models/comments.models");
+
+exports.getAllCommentsByArticleId = (request, response, next) => {
+    const { article_id } = request.params;
```
Comment (B7Bprog) [inline controllers/comments.controllers.js:5]: You will need to check the reason for the query returning empty rows:
- article exists but has no comments.
- article does not exist.

You will need to check whether the article exists in order to determine this.
I recommend the use of Promise.all() here.

**Review by B7Bprog — CHANGES_REQUESTED** [review]
Hey Isabel, please see my comments and refactor your code, then please send us another PR with it.

**Review by B7Bprog — APPROVED** [review]
Hey Isabel, nice work on refactoring your code. Please merge! 🎉
